server/src/main/java/org/eclipse/lsp/cobol/service/delegates/validations/CobolLanguageEngineFacade.java

```diff
@@ -111,7 +111,7 @@ private AnalysisResult toAnalysisResult(ResultWithErrors<SemanticContext> result
 
   private List<CodeBlockDefinitionNode> collectCodeBlockNodes(
       SemanticContext context, NodeType nodeType) {
-    return context
+    return Objects.isNull(context.getRootNode()) ? emptyList() :context
```

Comment:
seems visitor.visit gives an empty tree, i.e. no root node. When we are in between the syntax for copybook, whose copybook processing couldn't be completed by preprocessor yet.